base/compiler/typelimits.jl

```diff
@@ -540,24 +545,27 @@ function tmerge(lattice::PartialsLattice, @nospecialize(typea), @nospecialize(ty
                 typea.parent === typeb.parent)
                 return widenconst(typea)
             end
-            return PartialOpaque(typea.typ, tmerge(typea.env, typeb.env),
+            return PartialOpaque(typea.typ, tmerge(lattice, typea.env, typeb.env),
                 typea.parent, typea.source)
         end
         typea = aty
         typeb = bty
     elseif apo
-        typea = widenconst(typea)
+        typea = widenlattice(lattice, typea)
```

Comment:
```suggestion
        typea = widenlattice(wl, typea)
```
Otherwise `widenlattice(lattice, typea)` always returns `typea::PartialOpaque`?

Reply:
Good catch